File: src/main/java/com/esprit/models/RatingActivity.java
package com.esprit.models;

import javafx.beans.property.IntegerProperty;
import javafx.beans.property.SimpleIntegerProperty;

public class RatingActivity {
    private final IntegerProperty idActivity = new SimpleIntegerProperty(this, "idActivity");
    private final IntegerProperty idSalle = new SimpleIntegerProperty(this, "idSalle");
    public static final int ID_CLIENT_FIXE = 2; // ID client par défaut pour les tests
    private final int idClient;
    private int ratingStars;
    private String review;

    // ✅ Constructeur principal
    public RatingActivity(int idActivity, int idSalle, int idClient, int ratingStars, String review) {
        setIdActivity(idActivity);
        setIdSalle(idSalle);
        this.idClient = idClient; // Permet d'avoir un client dynamique
        this.ratingStars = ratingStars;
        this.review = review;
    }

    // ✅ Constructeur alternatif (avec ID client fixe par défaut)
    public RatingActivity(int idActivity, int idSalle, int ratingStars, String review) {
        this(idActivity, idSalle, ID_CLIENT_FIXE, ratingStars, review);
    }

    public int getIdActivity() {
        return idActivity.get();
    }

    public void setIdActivity(int idActivity) {
        this.idActivity.set(idActivity);
    }

    public IntegerProperty idActivityProperty() {
        return idActivity;
    }

    public int getIdSalle() {
        return idSalle.get();
    }

    public void setIdSalle(int idSalle) {
        this.idSalle.set(idSalle);
    }

    public IntegerProperty idSalleProperty() {
        return idSalle;
    }

    public int getIdClient() {
        return idClient;
    }

    public int getRatingStars() {
        return ratingStars;
    }

    public void setRatingStars(int ratingStars) {
        this.ratingStars = ratingStars;
    }

    public String getReview() {
        return review;
    }

    public void setReview(String review) {
        this.review = review;
    }

    @Override
    public String toString() {
        return "RatingActivity{" +
                "idActivity=" + getIdActivity() +
                ", idSalle=" + getIdSalle() +
                ", idClient=" + getIdClient() +
                ", ratingStars=" + ratingStars +
                ", review='" + review + '\'' +
                '}';
    }
}

File: src/Entity/Ratingactivity.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use App\Entity\Client;

#[ORM\Entity]
class Ratingactivity
{

    #[ORM\Id]
        #[ORM\ManyToOne(targetEntity: Activity::class, inversedBy: "ratingactivitys")]
    #[ORM\JoinColumn(name: 'idActivity', referencedColumnName: 'idActivity', onDelete: 'CASCADE')]
    private Activity $idActivity;

    #[ORM\Id]
        #[ORM\ManyToOne(targetEntity: Sallesportif::class, inversedBy: "ratingactivitys")]
    #[ORM\JoinColumn(name: 'idSalle', referencedColumnName: 'idSalle', onDelete: 'CASCADE')]
    private Sallesportif $idSalle;

    #[ORM\Column(type: "integer")]
    private int $ratingStars;

    #[ORM\Column(type: "string", length: 250)]
    private string $review;

    #[ORM\Id]
        #[ORM\ManyToOne(targetEntity: Client::class, inversedBy: "ratingactivitys")]
    #[ORM\JoinColumn(name: 'idClient', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private Client $idClient;

    public function getIdActivity()
    {
        return $this->idActivity;
    }

    public function setIdActivity($value)
    {
        $this->idActivity = $value;
    }

    public function getIdSalle()
    {
        return $this->idSalle;
    }

    public function setIdSalle($value)
    {
        $this->idSalle = $value;
    }

    public function getRatingStars()
    {
        return $this->ratingStars;
    }

    public function setRatingStars($value)
    {
        $this->ratingStars = $value;
    }

    public function getReview()
    {
        return $this->review;
    }

    public function setReview($value)
    {
        $this->review = $value;
    }

    public function getIdClient()
    {
        return $this->idClient;
    }

    public function setIdClient($value)
    {
        $this->idClient = $value;
    }
}
